.github/workflows/client.yml

```diff
@@ -0,0 +1,81 @@
+name: Go
+
+#TODO: extend the conditions once workflow gets tested together with other workflows
+on:
+  push:
+    branches:
+      # TODO: Run on master after we're fully migrated from Circle CI
+      # - master
+      - "rfc-18/**"
+  pull_request:
+      branches:
+      # TODO: Run on all branches after we're fully migrated from Circle CI
+      - "rfc-18/**"
```

Comment:
We should probably add exclude paths for dirs like `docs`, `infrastructure`, and `staker-rewards`. But let's do it in the future, not in this PR.